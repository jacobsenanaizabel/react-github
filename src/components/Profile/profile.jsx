import React from 'react'
import { Link } from "react-router-dom"
import axios from 'axios'

import './profile.scss'

export default class Profile extends React.Component{

    constructor(props) {
        super(props);
        this.state = {
            urlUser : this.props.location.state.profile,
            userInfo : []
        }
    }

    componentDidMount() {
        axios.get(this.state.urlUser, { crossdomain: true } )
            .then(response => {
                this.setState({userInfo : [response.data]});
            }
        ).catch(function (error) {
            console.error('erro no request :',error);
        });
    }

    render() {
        return(     
            <div className="user-container">
             {
                this.state.userInfo.map(user => 
                <div className="container-user">
                    <img src={user.avatar_url} class="user-avatar" alt="Cinque Terre" width="304" height="236"/>
                    <label><h2 className="user-name">{user.name}</h2></label>
                    <label><p className="user-location">{user.location}</p> </label>
                    <label><p className="user-email">{user.email}</p></label>
                    <textarea readonly class="form-control" rows="5" id="comment">
                        {user.bio}
                    </textarea>
                    <button type="button" class="btn btn-primary repository-button">
                    <Link className="text-white" to={{pathname: '/repository',
                                                state: { repository: user.repos_url } }}> Reposit√≥rios </Link>
                    </button>
                    <ul>                    
                        <li class="user-follow">Seguidores {user.followers}</li>
                        <li class="user-follow">Seguindo {user.following} </li>
                    </ul>
                </div>                         
             )}       
  
            </div>
        )
    }
}