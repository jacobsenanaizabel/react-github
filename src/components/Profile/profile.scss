.user-container {
    width: 60%;
    margin-left: 18%;
    background-color: #fafbfc;
    font-family: 'Roboto Condensed';

    .container-user {
        display: flex;
        position: relative;
        flex-direction: column;
        justify-content: center;
        align-items: center;

        img.user-avatar {
            width: 25%;
            height: 210px;
            border-radius: 100px;
            margin-top: 5%;
        }

        .user-name {
            font-size: 50px;
            margin: 0px;
        }

        .user-location {
            font-size: 25px;
        }

        .repository-button {
            width: 43%;
            height: 53px;
            margin: 5px;
        }
        
        > textarea {
            background-color: #fafbfc;
            width: 43%;
            border: none;
            overflow: auto;
            font-size: 18px;
        }
        
        > li {
            font-size: 25px;
        }

    }
}