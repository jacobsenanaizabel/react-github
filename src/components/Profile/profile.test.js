import React from 'react'
import { shallow } from 'enzyme'
import { MemoryRouter } from "react-router-dom"
import '../../setupTests'
import nock from 'nock'

import Profile from './profile'

describe('<Profile />', () => {

    it('renders user profile', () => {
        let wrapper = shallow(
            <MemoryRouter initialEntries={[ '/profile', { user: 'jacobsenanaizabel' } ]} initialIndex={0}>
                <Profile/>
            </MemoryRouter>
        );
    })

    it('return list users', () => {

        let user = 'jacobsenanaizabel';

        var userResponse = 
        {
            "login": "jacobsenanaizabel",
            "id": 13037895,
            "node_id": "MDQ6VXNlcjEzMDM3ODk1",
            "avatar_url": "https://avatars1.githubusercontent.com/u/13037895?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/jacobsenanaizabel",
            "html_url": "https://github.com/jacobsenanaizabel",
            "followers_url": "https://api.github.com/users/jacobsenanaizabel/followers",
            "following_url": "https://api.github.com/users/jacobsenanaizabel/following{/other_user}",
            "gists_url": "https://api.github.com/users/jacobsenanaizabel/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/jacobsenanaizabel/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/jacobsenanaizabel/subscriptions",
            "organizations_url": "https://api.github.com/users/jacobsenanaizabel/orgs",
            "repos_url": "https://api.github.com/users/jacobsenanaizabel/repos",
            "events_url": "https://api.github.com/users/jacobsenanaizabel/events{/privacy}",
            "received_events_url": "https://api.github.com/users/jacobsenanaizabel/received_events",
            "type": "User",
            "site_admin": false,
            "name": "Ana Izabel Jacobsen",
            "company": null,
            "blog": "https://www.facebook.com/jacobsenana",
            "location": "Rio de Janeiro ",
            "email": null,
            "hireable": null,
            "bio": null,
            "public_repos": 29,
            "public_gists": 0,
            "followers": 14,
            "following": 23,
            "created_at": "2015-06-24T18:21:55Z",
            "updated_at": "2018-07-02T17:12:45Z"
        }

        nock('https://api.github.com')
            .get('/users/'+ user)
            .reply(200, userResponse);
    })

})