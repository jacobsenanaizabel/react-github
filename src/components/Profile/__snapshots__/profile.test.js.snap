// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders repository snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "pr2wff",
          "pathname": "/profile",
          "search": "",
          "state": undefined,
        },
        Object {
          "hash": "",
          "key": "5srd53",
          "pathname": "/",
          "search": "",
          "user": "jacobsenanaizabel",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 2,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "pr2wff",
        "pathname": "/profile",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Profile />
</Router>
`;
